central/resourcecollection/datastore/datastore_impl.go

```diff
@@ -427,3 +431,161 @@ func verifyCollectionObjectNotEmpty(obj *storage.ResourceCollection) error {
 	}
 	return nil
 }
+
+func (ds *datastoreImpl) ResolveListDeployments(ctx context.Context, collection *storage.ResourceCollection) ([]*storage.ListDeployment, error) {
+
+	if err := verifyCollectionConstraints(collection); err != nil {
+		return nil, err
+	}
+
+	// currently we only support one resource selector per collection from UX
+	if collection.GetResourceSelectors() != nil && len(collection.GetResourceSelectors()) > 1 {
+		return nil, errors.New("only 1 resource selector is supported per collection")
+	}
```

Comment:
This is already covered by `verifyCollectionConstraints`. Refer to my comment on simplifying validation code.